src/augeas.c

```diff
@@ -1993,6 +1993,80 @@ int tree_equal(const struct tree *t1, const struct tree *t2) {
 }
 
 /*
+ * aug_ns_* functions
+ */
+int aug_ns_get(augeas* aug, const char *var, int i, const char **value) {
+    int result = -1;
+
+    if (value != NULL)
+        *value = NULL;
+
+    api_entry(aug);
+
+    struct tree *tree = pathx_symtab_get_tree(aug->symtab, var, i);
+    ERR_THROW(tree == NULL, aug, AUG_ENOMATCH,
+              "Node %s[%d] does not exist", var, i);
+
+    if (value != NULL)
+        *value = tree->value;
+
+    result = 0;
```

Comment:
Result doesn't match how aug_get() returns it's result. I expected aug_ns_get to return 1 on success.
It might not make sense but it breaks user expectations.